esp-link/cgiwifi.c

```diff
@@ -281,239 +298,427 @@ int ICACHE_FLASH_ATTR cgiWiFiScan(HttpdConnData *connData) {
 #define RESET_TIMEOUT (15000) // 15 seconds
 static ETSTimer resetTimer;
 
-// This routine is ran some time after a connection attempt to an access point. If
-// the connect succeeds, this gets the module in STA-only mode. If it fails, it ensures
-// that the module is in STA+AP mode so the user has a chance to recover.
+/*
+ *  This routine is ran some time after a connection attempt to an access point.
+ *  If the connection succeeds and Makefile hard-coded CHANGE_TO_STA is set to yes, this gets the module in STA-only mode.
+ *  If it fails and only STA mode is set, it ensures that the module is in STA+AP mode so the user has a chance to recover.
+ */
 static void ICACHE_FLASH_ATTR resetTimerCb(void *arg) {
-  int x = wifi_station_get_connect_status();
-  int m = wifi_get_opmode() & 0x3;
-  DBG("Wifi check: mode=%s status=%d\n", wifiMode[m], x);
-
-  if (x == STATION_GOT_IP) {
-    if (m != 1) {
+    
+    int x = wifi_station_get_connect_status();
+    int m = wifi_get_opmode() & 0x3;
+    DBG("Wifi check: mode=%s status=%d\n", wifiMode[m], x);
+    
+    if(m!=2){
+        if( x == STATION_GOT_IP ){
 #ifdef CHANGE_TO_STA
-      // We're happily connected, go to STA mode
-      DBG("Wifi got IP. Going into STA mode..\n");
-      wifi_set_opmode(1);
-      os_timer_arm(&resetTimer, RESET_TIMEOUT, 0); // check one more time after switching to STA-only
+            wifi_set_opmode(1);
```

Comment:
Does this cause a hiccup if we're already in opmode==1? Would be safer to prefix with `if (opmode != 1)`